src/ws_server.cu

```diff
@@ -310,8 +314,13 @@ class AprilTagHandler : public seasocks::WebSocket::Handler {
         frame_counter++;
 
         auto overallstart = std::chrono::high_resolution_clock::now();
-        if (gui_rotate_img_) {
-          rotateImage(&bgr_img, 180.0);
+        //Let's check the time this takes, can always combine to one call if both are true later.
+        //Best case scenario is we don't place the camera wrong so we do not need this method at all.
+        if (flipVertical_) {
+          flipVertical(&bgr_img);
+        }
+        if (flipHorizontal_) {
+          flipHorizontal(&bgr_img);
```

Comment:
same comment as above here.

Reply:
I did some experiments last night to figure out what was going on.  The results are in this truth table:

| tag detected | vertical flip | horizontal flip |
| -------------- | ------------ | --------------- |
| true               |   false         |        false         |
| false              |   true          |        false         |
| false              |   false         |        true          |
| true               |   true          |        true          |

If no checkboxes are clicked the tag is detected.  If one checkbox is clicked and the other one is not, the tag is not detected, but if both of them are clicked the tag is detected.

It took me a few minutes to figure out why!  I had to print out an apriltag and flip it and rotate it in various ways to convince myself of what was happening.  Basically a mirror image of an apriltag is not an apriltag pattern.  but a mirror image of a mirror image is.  So checking both boxes is equivalent to a 180 degree rotation, while only clicking one box is just a reflection.

Think of it this way - when you take an apriltag and flip it vertically with cv::flip, you're actually looking at it upside down from behind.  It's as if you took the apriltag and flipped it down on the bottom edge and now you're looking at the back of the apriltag.  It;'s upside down and backwards from the original tag. 
 And when you apply the horizontal flip, now you're flipping the tag over along the right hand edge.  So now you are looking at the front again, and it is upside down from it's original orientation, so its equivalent to a 180 degree rotation.

You might have to print this out and try it!

So the fix here is to have one call to cv::flip which uses the flip code of -1 to do both directions simultaneously.  Does this make sense?  We can do a video meet and discuss if you like, or I can show you in person at the next robotics session.

Reply:
Makes perfect sense. I can sort of visualize it in my head! I just pushed a fix for the -1 option and will look into what P.V. does for their other 90 degree rotations

Reply:
this is looking good now. what do you want to do about the ui options?  I'm concerned that if we leave the two options on there that we will get to a competition and forget to click both of them because we'll be tired and then the vision system won't work and we'll be scrambling.  How about just a single option for now and we figure out 90 degree rotation later if we need it?

Reply:
Going to leave it as two for now but both are required to be checked for it to work. Then, we can leave it for easy dev later. I'm also going to add a warning in the UI that both must be clicked.

Reply:
Also -- best case scenario is we place the cameras correctly so we don't even need to worry about this feature :)